crates/burn-import/src/onnx/op_configuration.rs

```diff
@@ -209,6 +210,75 @@ pub fn max_pool2d_config(curr: &Node) -> MaxPool2dConfig {
         .with_padding(padding)
         .with_dilation([dilations[0] as usize, dilations[1] as usize])
 }
+
+pub fn conv_transpose1d_config(curr: &Node) -> ConvTranspose1dConfig {
+    let mut attrs = curr.attrs.clone();
+
+    // Extract kernel_shape, default to an empty vector if not present
+    let kernel_shape = attrs
+        .remove("kernel_shape")
+        .map(AttributeValue::into_i64s)
+        .unwrap_or_default();
+
+    // Extract strides, default to 1 if not present
+    let stride = attrs
+        .remove("strides")
+        .map(AttributeValue::into_i64s)
+        .unwrap_or_else(|| vec![1]);
+
+    // Extract padding, default to 0 if not present
+    let pads = attrs
+        .remove("pads")
+        .map(AttributeValue::into_i64s)
+        .unwrap_or_else(|| vec![0, 0]); // Begin and end padding
+
+    // Extract dilations, default to 1 if not present
+    let dilations = attrs
+        .remove("dilations")
+        .map(AttributeValue::into_i64s)
+        .unwrap_or_else(|| vec![1]);
+
+    // Extract group attribute, default to 1
+    let group = attrs
+        .remove("group")
+        .map(AttributeValue::into_i64)
+        .unwrap_or(1) as usize;
+
+    // Extract output_padding, default to 0 if not present
+    let output_padding = attrs
+        .remove("output_padding")
+        .map(AttributeValue::into_i64s)
+        .unwrap_or_else(|| vec![0]);
+
+    // Ensure no unused attributes remain
+    if !attrs.is_empty() {
+        panic!("Not all attributes are used: {attrs:?}");
+    }
+
+    // Extract weight tensor, verify it's present
+    let weight = if let ArgType::Tensor(ref weight) = curr.inputs[1].ty {
+        weight
+    } else {
+        panic!("ConvTranspose1d: weight tensor must be present");
+    };
+
+    // Check if bias is present (third input)
+    let bias = curr.inputs.len() == 3;
+
+    // Extract channels from the weight tensor shape [out_channels, in_channels]
+    let shape = weight.shape.clone().unwrap();
+    let channels: [usize; 2] = [shape[1] * group, shape[0]];
+
+    // Create the ConvTranspose1d configuration
+    ConvTranspose1dConfig::new(channels, kernel_shape[0] as usize)
+        .with_stride(stride[0] as usize)
+        .with_padding(pads[0] as usize) // Only use the first padding value for 1D
```

Comment:
Actually, your changes along with the commend here made me realize that the padding is not handled correctly for conv transpose ops with ONNX import.

ONNX defines `pads` as `[x1_begin, x2_begin...x1_end, x2_end,...]`, where `xi_begin` the number of pixels added at the beginning of axis `i` and `xi_end`, the number of pixels added at the end of axis `i`.

But we only support symmetric padding (`xi_begin` = `xi_end`), that's why the [padding for `ConvTranspose2d`](https://github.com/tracel-ai/burn/blob/main/crates/burn-core/src/nn/conv/conv_transpose2d.rs#L68) only takes two values (one for each dimension, symmetric on both sides) and `ConvTranspose1d` only takes one value.

We need to check that the values defined are symmetric, similar to how it's checked for convolution ops (e.g., [`padding_config_conv2d`](https://github.com/tracel-ai/burn/blob/main/crates/burn-import/src/onnx/op_configuration.rs#L1000) - but only check that the values are not negative or asymmetric).

If you want to fix the other conv transpose ops in this PR feel free to 🙂 otherwise I can tackle this in a separate PR after this one.

Reply:
@laggui I appreciate your feedback and review. I will take the padding issue, so give me some time to look into it.